Source/WebCore/testing/Internals.cpp

```diff
@@ -6827,4 +6827,20 @@ String Internals::modelInlinePreviewUUIDForModelElement(const HTMLModelElement&
 
 #endif
 
+void Internals::avoidIOSurfaceSizeCheckInWebProcess(Element& element)
+{
+    auto* document = contextDocument();
+    if (!document)
+        return;
+    auto* page = document->page();
+    if (!page)
+        return;
+    page->settings().setMaximumAccelerated2dCanvasSize(UINT_MAX);
+    HTMLCanvasElement::setMaxCanvasAreaForTesting(UINT_MAX);
+    HTMLCanvasElement::setMaxPixelMemoryForTesting(UINT_MAX);
+    if (is<HTMLCanvasElement>(element))
+        downcast<HTMLCanvasElement>(element).setAvoidIOSurfaceSizeCheckInWebProcessForTesting();
```

Comment:
Since this API is not going to work correctly if the element is not `HTMLCanvasElement`, why do not we pass an `HTMLCanvasElement` instead of passing `Element`?

Reply:
Fixed.